packages/good-design/src/core/layout/ClaimCarousel.tsx

```diff
@@ -0,0 +1,98 @@
+import { FlatList, View } from "native-base";
+import React, { FC, useCallback, useMemo, useState } from "react";
+import { LayoutChangeEvent, NativeScrollEvent, NativeSyntheticEvent } from "react-native";
+import { IClaimCard } from "../buttons";
+import ClaimCard from "./ClaimCard";
+
+interface ClaimCarouselProps {
+  cards: Array<IClaimCard>;
+}
+
+const ClaimCarousel: FC<ClaimCarouselProps> = ({ cards }) => {
+  const [slidesNumber, setSlidesNumber] = useState(0);
+  const [activeSlide, setActiveSlide] = useState(0);
+
+  const onFlatListLayoutChange = useCallback(
+    (event: LayoutChangeEvent) => {
+      const contentWidth = cards.length * 275 + (cards.length - 1) * 16;
+
+      if (event.nativeEvent.layout.width >= contentWidth) {
+        setSlidesNumber(0);
+        return;
+      }
+
+      setSlidesNumber(Math.ceil((contentWidth - event.nativeEvent.layout.width - 16) / (275 + 16)));
+    },
+    [cards, setSlidesNumber]
+  );
+
+  const onScroll = useCallback(
+    (event: NativeSyntheticEvent<NativeScrollEvent>) => {
+      const currentSlide = Math.floor(event.nativeEvent.contentOffset.x / (275 + 16));
+
+      if (activeSlide === currentSlide) return;
+
+      setActiveSlide(currentSlide);
+    },
+    [activeSlide, setActiveSlide]
+  );
+
+  const slidesComponent = useMemo(
+    () =>
+      Array(slidesNumber)
+        .fill(0)
+        .map((_, index, arr) => (
+          <View
```

Comment:
create separate component wrapped onto React.memo

Reply:
why do you need both slides views and flatlist ?

Reply:
this is for bottom pagination